server/api.js

```diff
@@ -58,122 +60,151 @@ router.get("/callback", async (req, res) => {
     const { data } = await octokit.request("/user");
     return res.status(200).json(data.login);
   } catch (error) {
-    console.error("Access Token Error", error.message);
     return res.status(500).json("Authentication failed");
   }
 });
 
-router.get("/", (_, res, next) => {
-  Connection.connect((err) => {
-    if (err) {
-      return next(err);
-    }
-    res.json({ message: "Hello, world!" });
-  });
-});
-router.get("/graduates", (_, res, next) => {
-  Connection.connect((err) => {
-    if (err) {
-      return next(err);
-    }
-    Connection.query("SELECT * FROM graduates", (error, result) => {
+router.get("/graduates", (req, res) => {
+  Connection.query("SELECT * FROM graduates", (error, result) => {
+    if (result) {
       res.json(result.rows);
-    });
+    } else {
+      res.send(error);
+    }
   });
 });
+
 // create new profile
 router.post("/graduates", function (req, res) {
   const newFirstName = req.body.first_name;
   const newSurname = req.body.surname;
-  const github_name = req.body.githubName;
-  //checking if the user is existed in our github table
+  const aboutMe = req.body.about_me;
+  const location = req.body.location;
+  const interest=req.body.interest;
+  const github=req.body.github_link;
+  const linkedin=req.body.linkedin_link;
+  const portfolio=req.body.portfolio_link;
+  const github_id = req.body.github_id;
+  const avatar_url=req.body.avatar_url;
+  const skills =req.body.skills.map(x=>x.toLowerCase());
+
+  Connection.query(
+          `insert into graduates (first_name, surname, about_me, location, interest, github_link, linkedin_link, portfolio_link, github_id, avatar_url ) values` +
+            `($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) returning *`,
+          [newFirstName, newSurname, aboutMe, location, interest, github, linkedin, portfolio, github_id, avatar_url],
+          (error, result) => {
+            if(result){
+              let graduate_id=result.rows[0].id
+              Connection.query(
+                `insert into graduate_skill (graduate_id, skill_id)` +
+                 ` select $1, id from skills where skill_name=ANY($2)`, [graduate_id, skills],
+                (err,result)=>{
+                  if (!err){
```

Comment:
would be better to check the error everywhere instead checking the result sometimes here and there: see line 96 for example